### PR TITLE
fix backward incompatibility of scapy.utils.RawPcapReader.read_packet() for scapy's v2.4 againts v2.3.2 & v2.3.3

### PR DESCRIPTION
**Pull request by karpierz**
See: #1128

### CODE REVIEW

File: scapy/utils.py [hunk 1]
```diff
@@ -816,7 +817,6 @@ def next(self):
         return pkt
     __next__ = next
 
-
```
Comment (p-l-) [inline scapy/utils.py]: Please avoid committing unnecessary changes.

File: scapy/utils.py [hunk 2]
```diff
@@ -827,8 +827,10 @@ def read_packet(self, size=MTU):
             return None
         sec,usec,caplen,wirelen = struct.unpack(self.endian+"IIII", hdr)
         s = self.f.read(caplen)[:size]
-        return s, (sec, usec, caplen, wirelen)
+        return s, self._PacketInfo(sec=sec, usec=usec, wirelen=wirelen, caplen=caplen)
 
+    _PacketInfo = collections.namedtuple("PacketInfo",
```
Comment (p-l-) [inline scapy/utils.py]: Please move this declaration **before** its first use, for readability. Also, since this type will be used for values returned by a public method, I don't think it should be private. Could you change this name to `PacketInfo` or even better `PacketMetadata`?

Comment (gpotter2) [reply scapy/utils.py]: `PacketMetadata` is confusing: we already have `Packet_metaclass`
`PacketInfo` or anything similar is good to me

Comment (p-l-) [reply scapy/utils.py]: I do not agree. Metaclass and Metadata have different meanings. Metadata seems more precise (and clear) than Info to me.

Comment (gpotter2) [reply scapy/utils.py]: Fair enough 👍

Comment (karpierz) [reply scapy/utils.py]: I'v done a changes. 
Is there a way to update this PR or I should create next one?
PS: I will deal with RawPcapNgReader, but in the next issue/PR.

**Comment by karpierz** [comment]
Ready to merge.

**Comment by karpierz** [comment]
> Also, don't you think the new type should be used in the appropriate RawPcapNgReader.read_block*()
> and in PcapNgReader.read_packet()?
Its done in the last commit in this PR.

File: scapy/utils.py [hunk 3]
```diff
@@ -816,6 +817,8 @@ def next(self):
         return pkt
     __next__ = next
 
+    PacketMetadata = collections.namedtuple("PacketInfo",
```
Comment (p-l-) [inline scapy/utils.py]: The types (both this one and `BlockMetadata`) should not be related to the object (sorry I had missed that). Could you declare them outside the class declaration? Also, could you set their name to "PacketMetadata" and "BlockMetadata"?

Comment (karpierz) [reply scapy/utils.py]: Already fixed.

Comment (karpierz) [reply scapy/utils.py]: ..but what about RawPcapNgReader.BlockMetadata name which I used?
Maybe better if will be the same (PacketMetadata) as in RawPcapReader ?

Comment (karpierz) [reply scapy/utils.py]: :: The types (both this one and BlockMetadata) should not be related to the object (sorry I had missed that).

Cannot agree. Its are used only in this class (as a result only this class methods), so it is specific for this class/type only.
This idiom is very often (especially because Python has no namespaces) and quite correct.
Using such names in the module namespace/scope has less sense.
Much better is:
RawPcapReader.BlockMetadata than RawPcapReader_BlockMetadata and
RawPcapNgReader.BlockMetadata than RawPcapNgReader_BlockMetadata

**Comment by karpierz** [comment]
Could you check (and restar/resume) travis jobs?
Weird error appears. eg.g OpenSSL certificate problem, fro now problem with threads and similar not related to PR changes.
I do not have the strength and time to fight him:(

**Comment by karpierz** [comment]
PR Ready for merge

**Comment by p-l-** [comment]
@karpierz sorry to bother you with that, but the object types should be declared outside the object, since they are used in valued that can be **returned** by non-private methods, which means the type can be used in other places.

Also, you may want to rebase your changes against current master, remove the "merge" commit and clean-up the commit history. What do you think?

**Comment by karpierz** [comment]
Still cannot agree (treat our talk as normal discussion - not as any kind of fighting etc:) 
Inner classes/types  exists in several prog.langs (e.g. Java) and of course exist as normal
and recomended technique and can be used as normal public classes/types.
The difference is only the place of declaration/namespace (Java has no namespace 
construct in the similar way as Python - just inner classes replace/covers this role).
So you _always_ can use normal construct like:
a = module.MainClass.InnerClass(args)
It is more natural than:
a = module.MainClass_InnerClass(args)
and e.g. without unnecessary littering of module namespace/__dict__ etc.

PS: Of course I will not insist if you do not agree, but I sincerely recommend you/and others
just this technique (inner classes).

Of course I will try to clean commits etc - get me plesa a day/two :) (I am on vacation:)

**Comment by karpierz** [comment]
Ready for merge to the master

**Comment by karpierz** [comment]
PR is ready for merge.

**Comment by karpierz** [comment]
Please help :)
Could you restart a failed job/task on travis?

**Comment by karpierz** [comment]
PR is ready for merge.

**Comment by karpierz** [comment]
Leave a comment

**Comment by karpierz** [comment]
PR is ready for merge.

**Comment by p-l-** [comment]
@karpierz I know you're going to call me an idiot (and you'd be right) but I was away from Scapy for some days (sorry about that), and I hadn't read your comment. I just did, and I **do** agree with you with this explanation. 

So, sorry to ask that, but can you write the code as it was previously? I'm really sorry to bother you, your work on that is great and I really appreciate it.

**Comment by karpierz** [comment]
:: I know you're going to call me an idiot (and you'd be right)

Hehe not quite ;). In this particular place (using of named tuple) "unfortunately" you was right :).
If  we use named tuple in "my" way, the name of namedtuple (first param)  cannot contain dot.
so does not corresponds correctly with its namespace (class). So it is wrong and your way is more correct.

PS: In general cases (when we declare usual classes as subtypes) I still (will) prefer "my" way ;)

So PR is fully ready for merge.

**Comment by p-l-** [comment]
OK! Thanks a lot for your work!
